File: growing-shop-audience/src/main/java/com/example/growingshopaudience/category/domain/Category.java
package com.example.growingshopaudience.category.domain;

import com.example.growingshopaudience.category.dto.CategoryRequest;
import com.example.growingshopcommon.validator.StringChecker;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @NotBlank
    @Size(max = 100)
    @StringChecker(includeLowerEn = true, includeUpperEn = true, includeKo = true, includeNumber = true, includeSpecialCharacter = "-.\"'")
    private String name;

    @Enumerated(EnumType.STRING)
    private CategoryStatus status;

    public Category(String name) {
        this.name = name;
        this.status = CategoryStatus.UNDER_REVIEW;
    }

    public void update(CategoryRequest.CategoryReq req) {
        this.name = req.getName();
    }

    public void delete() {
        this.status = CategoryStatus.DELETED;
    }

    public void reject() {
        this.status = CategoryStatus.REJECTED;
    }

    public void confirm() {
        this.status = CategoryStatus.NORMAL;
    }
}

File: growing-shop-domain/src/main/kotlin/com/example/domain/category/Category.kt
package com.example.domain.category

import com.example.domain.common.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

@Entity
class Category(
    @Column(nullable = false)
    @NotBlank
    @Size(max = 100)
    @Pattern(regexp = "^[a-zA-Z가-힣-.\"']+$")
    val name: String,

    status: CategoryStatus
) : BaseEntity() {
    @Enumerated(EnumType.STRING)
    var status: CategoryStatus = status
        protected set

    fun delete() {
        status = CategoryStatus.DELETED
    }

    fun inspect(status: CategoryStatus) {
        when (status) {
            CategoryStatus.REJECTED -> this.status = CategoryStatus.REJECTED
            CategoryStatus.NORMAL -> this.status = CategoryStatus.NORMAL
            else -> throw IllegalArgumentException("잘못된 검수 요청입니다.")
        }
    }
}
